spec/system/potential_user_browses_breweries_spec.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "spec_helper"
+
+RSpec.feature "A potential user sends an external request to search breweries" do
+  scenario "with valid parameters in the OpenBrewery DB Search feature" do
+    visit "/"
```

Comment:
Before you navigate to the app is where I would suggest making Mocks and stubbing methods. I havent looked at this code deeply, but there is where you may want to stub the response of your service object. At the time your `click_on` method gets called, nothing is stubbed, so actual code gets executed. It probably doesnt work as expected in the test suite, and likely doesnt send the right data to the page. Either that, or it actually does work, and your test suite is making request/response cycles to the internet. 

Lets chat about the 4 phases of testing today: https://thoughtbot.com/blog/four-phase-test. It will likely clarify when/where you want to do any stubbing. 

As for your question below, you _could_ stub the response of `connection.get`, but you could also stub out calls "further up the call stack" and just stub the service object's call. That way the client component wont even be involved.

Reply:
Right. I think I addressed this properly in https://github.com/orhanna14/homebrew-buddy/pull/25. Let me know if not.